editor/plugins/inspector_action_editor_plugin.cpp

```diff
@@ -0,0 +1,86 @@
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "inspector_action_editor_plugin.h"
+#include "editor/editor_node.h"
+#include "editor/editor_property_name_processor.h"
+
+bool EditorInspectorActionPlugin::can_handle(Object *p_object) {
+	Ref<Script> scr = Object::cast_to<Script>(p_object->get_script());
+	return scr.is_valid() && (*scr)->is_tool();
+}
+
+void EditorInspectorActionPlugin::update_action_icon(Button *action_button) {
+	action_button->set_icon(action_button->get_theme_icon(action_icon, SNAME("EditorIcons")));
+}
+
+void EditorInspectorActionPlugin::call_action(Object *p_object, String p_path) {
+	Variant ret;
+	p_object->get_script_instance()->get(StringName(p_path), ret);
+
+	Callable *callable = VariantGetInternalPtr<Callable>::get_ptr(&ret);
+
+	if (callable->is_valid()) {
+		Callable::CallError ce;
+		Variant v;
+		callable->call(nullptr, 0, v, ce);
+		ERR_FAIL_COND_MSG(ce.error != Callable::CallError::CALL_OK, "Could not call action. Error: " + Variant::get_callable_error_text(*callable, nullptr, 0, ce));
+	} else {
+		print_error("Could not call action. Callable invalid.");
+	}
+}
+
+bool EditorInspectorActionPlugin::parse_property(Object *p_object, const Variant::Type p_type, const String &p_path, const PropertyHint p_hint, const String &p_hint_text, const uint32_t p_usage, const bool p_wide) {
+	if (p_type == Variant::Type::CALLABLE) {
+		if (!p_hint_text.begins_with("action")) {
+			return false;
+		}
+		Button *action_button = EditorInspector::create_inspector_action_button(EditorPropertyNameProcessor::get_singleton()->process_name(p_path, EditorPropertyNameProcessor::STYLE_CAPITALIZED));
+
+		String icon = p_hint_text.substr(6);
+		if (!icon.is_empty()) {
+			action_icon = StringName(icon);
+			action_button->connect(SNAME("theme_changed"), callable_bind(callable_mp(this, &EditorInspectorActionPlugin::update_action_icon), action_button));
+			action_button->connect(SNAME("tree_entered"), callable_bind(callable_mp(this, &EditorInspectorActionPlugin::update_action_icon), action_button));
+		}
+		add_custom_control(action_button);
+
+		action_button->connect(SNAME("pressed"), callable_bind(callable_mp(this, &EditorInspectorActionPlugin::call_action), p_object, Variant(p_path)));
+		return true;
+	} else {
+		return false;
+	}
```

Comment:
We tend to use early return in this kind of situation to avoid extra indentation:
```suggestion
	if (p_type != Variant::Type::CALLABLE) {
		return false;
	}

	if (!p_hint_text.begins_with("action")) {
		return false;
	}
	Button *action_button = EditorInspector::create_inspector_action_button(EditorPropertyNameProcessor::get_singleton()->process_name(p_path, EditorPropertyNameProcessor::STYLE_CAPITALIZED));

	String icon = p_hint_text.substr(6);
	if (!icon.is_empty()) {
		action_icon = StringName(icon);
		action_button->connect(SNAME("theme_changed"), callable_bind(callable_mp(this, &EditorInspectorActionPlugin::update_action_icon), action_button));
		action_button->connect(SNAME("tree_entered"), callable_bind(callable_mp(this, &EditorInspectorActionPlugin::update_action_icon), action_button));
	}
	add_custom_control(action_button);

	action_button->connect(SNAME("pressed"), callable_bind(callable_mp(this, &EditorInspectorActionPlugin::call_action), p_object, Variant(p_path)));
	return true;
```